lib/Haskell/Law/Applicative/Def.agda

```diff
@@ -0,0 +1,37 @@
+module Haskell.Law.Applicative.Def where
+
+open import Haskell.Prim
+open import Haskell.Prim.Functor
+
+open import Haskell.Prim.Applicative
+open import Haskell.Prim.Monoid
+
+open import Haskell.Law.Functor
+
+record IsLawfulApplicative (func : Set → Set) ⦃ iAppF : Applicative func ⦄ : Set₁ where
+  field
+    overlap ⦃ super ⦄ : IsLawfulFunctor func
+
+    -- Identity: pure id <*> v = v
+    identity : (v : func a) → (pure id <*> v) ≡ v
+
+    -- Composition: pure (.) <*> u <*> v <*> w = u <*> (v <*> w)
+    composition : {a b c : Set} → (u : func (b → c)) (v : func (a → b)) (w : func a)
```

Comment:
Weirdly, for a lot of these, I can't remove them because other wise I either get an "Failed to solve the following constraints" error or have to define it like this:

```agda
Applicative.pure iAppF id
```

Reply:
Yeah, sometimes generalized variables mess up things, I think it's fine to leave this as is.